lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -390,36 +400,33 @@ def modify_cluster(module, redshift):
               'allow_version_upgrade', 'number_of_nodes', 'new_cluster_identifier',
               'enhanced_vpc_routing'):
         if p in module.params:
-            params[p] = module.params.get(p)
+            # https://github.com/boto/boto3/issues/400
+            if module.params.get(p) is not None:
+                params[p] = module.params.get(p)
 
     try:
-        redshift.describe_clusters(identifier)['DescribeClustersResponse']['DescribeClustersResult']['Clusters'][0]
-    except boto.exception.JSONResponseError as e:
+        redshift.describe_clusters(ClusterIdentifier=identifier)['Clusters'][0]
+    except botocore.exceptions.ClientError as e:
         try:
-            redshift.modify_cluster(identifier, **params)
-        except boto.exception.JSONResponseError as e:
-            module.fail_json(msg=str(e))
-
+            redshift.modify_cluster(ClusterIdentifier=identifier,
```

Comment:
I'm not sure why an error describing the cluster leads to trying to modify the cluster. What is the exception expected [here](https://github.com/ansible/ansible/pull/37052/files#diff-fd4728adea42973936ca9bdb0082deadR409)?